src/ECL/ECL.hs

```diff
@@ -43,15 +62,32 @@ data Guard = CatchAll | Opaque
 -- a collection of values, and an optional guard.
 type Alternative lit = (Binders lit, Maybe Guard)
 
--- | A list of uncovered cases
-newtype Uncovered lit = Uncovered { getUncovered :: [Binders lit] }
-  deriving (Show, Eq) 
+-- | Check wraps both uncovered and redundant cases.
+data Check lit = Check
+  { getUncovered :: [Binders lit]
+  , getRedundant :: (Maybe Bool, [Binders lit])
```

Comment:
I couldn't figure out how to do it. Do you mean something like `Maybe [Binders lit]`?
So, we can take the case `Just False` as `Just` some redundant list, the other cases don't carry information.
Also, I'm thinking that we can just remove `Maybe Bool` from that product, since we already used it in `missingSingle` and `missingMultiple` outside `Check` data-type, and doesn't provide useful information but that the right list in the product is non-empty.